### DISCUSSION TITLE
TomSelect element in shadow root

### DISCUSSION THREAD

**Discussion by jrief** [orchidjs/tom-select#132] [Q&A]
It seems, that when moving a **tom-select** element into the shadow root, some event are swallowed or somehow not propagated correctly.

For reference, I created a select element using the standard CSS as provided by [tom-select.css](https://github.com/orchidjs/tom-select/blob/master/dist/css/tom-select.css)
Here everything works as expected. Please check this [Code-Pen](https://codepen.io/jrief/pen/BaWbVVa).

Then I created the same example as before, but instead I moved the select element into a shadow root and initialized the tom-select element within. Since elements in a shadow root have to be styled isolated, I copied the CSS from above into a style-element and attached that inside the shadow root.
On the first sight, there seems to be no difference – what one would expect. Please check this [Code-Pen](https://codepen.io/jrief/pen/BaWbYxa).

However, some events seem to get swallowed. Clicking on an option, doesn't call the registered handler, ie. [addEvent(dropdown,'click',(evt) => {...}](https://github.com/orchidjs/tom-select/blob/690663262832d3d5bfa0db590d1cec05afe3001b/src/tom-select.ts#L288)
Also the behaviour when clicking on the input field for the first time, is different compared to the example above.

Any idea, what could cause these problems?

**Comment by oyejorge** [comment]
Interesting. I'm going to have to brush up on shadow root functionality but I wonder if it has something to do with [how we determine if a mousedown event is targeting an element inside the Tom Select instance](https://github.com/orchidjs/tom-select/blob/master/src/tom-select.ts#L333). Might be able to fix this by implementing [a contains() method like popper](https://github.com/popperjs/popper-core/blob/master/src/dom-utils/contains.js).

**Reply by oyejorge** [reply]
Just released 1.7.6 with the fix
